wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -573,6 +586,49 @@ private Optional<ExpressionFactory<String>> getExpressionFactory(RelationalTranf
     return ctx.getEngine().getExpressionFactory(StringExpressionFactoryType.SQL);
   }
 
+  /**
+   * This method emits all metrics for the given list of directives
+   *
+   * @param directives a list of Wrangler directives
+   * @param metrics CDAP {@link Metrics} object using which metrics can be emitted
+   * @throws DirectiveLoadException
+   */
+  private void emitDirectiveMetrics(List<Directive> directives, Metrics metrics) throws DirectiveLoadException {
+    for (Directive directive : directives) {
+      List<EntityMetricDef> entityMetricDefs = new ArrayList<>();
+
+      // Emit directive usage metric only if directive is present in system directive registry
```

Comment:
why only system ones?

Reply:
For now we want to emit usage metrics only for system directives and not for UDDs

Reply:
Currently we are focusing on usage metrics for system directives. We will add metrics for UDDs if required, in the next release.

Also, updated the logic to emit any metrics only if the directive is system directive.